src/fragments/lib/auth/ios/switch-auth/switch-auth.mdx

```diff
@@ -0,0 +1,84 @@
+
+3. `customWithSRP`: The `customWithSRP` flow is used to start with SRP authentication and then switch to custom authentication. This is useful if you want to use SRP for the initial authentication and then use custom authentication for subsequent authentication attempts.
+
+4. `customWithoutSRP`: The `customWithoutSRP` flow is used to start authentication flow **WITHOUT** SRP and then use a series of challenge and response cycles that can be customized to meet different requirements.
+
+`Auth` can be configured to use the different flows at runtime by calling `signIn` with `AuthSignInOptions`'s `authFlowType` as `AuthFlowType.userPassword`, `AuthFlowType.customAuthWithoutSrp` or `AuthFlowType.customAuthWithSrp`. If you do not specify the `AuthFlowType` in `AuthSignInOptions`, the default flow (`AuthFlowType.userSRP`) will be used.
+
+```swift 
+public enum AuthFlowType: String {
+
+    /// Authentication flow for the Secure Remote Password (SRP) protocol
+    case userSRP
+
+    /// Authentication flow for custom flow which are backed by lambda triggers.
+    /// Note that `custom`will always begin with a SRP flow.
+    @available(*, deprecated, message: "Use of custom is deprecated, use customWithSrp or customWithoutSrp instead")
+    case custom
+
+    /// Authentication flow which start with SRP and then move to custom auth flow
+    case customWithSRP
+
+    /// Authentication flow which starts without SRP and directly moves to custom auth flow
+    case customWithoutSRP
+
+    /// Non-SRP authentication flow; user name and password are passed directly.
+    /// If a user migration Lambda trigger is set, this flow will invoke the user migration
+    /// Lambda if it doesn't find the user name in the user pool.
+    case userPassword
+}
+```
+
+> For more information about authentication flows, please visit [AWS Cognito developer documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-authentication-flow.html#amazon-cognito-user-pools-custom-authentication-flow)
+
+## USER_PASSWORD_AUTH flow
+
+A use case for the `USER_PASSWORD_AUTH` authentication flow is migrating users into Amazon Cognito
+
+```swift
+func signIn(username: String, password: String) async throws {
+
+    let option = AWSAuthSignInOptions(authFlowType: .userPassword)
+    do {
+        let result = try await Amplify.Auth.signIn(
+            username: username,
+            password: password,
+            options: AuthSignInRequest.Options(pluginOptions: option))
+        print("Sign in succeeded with result: \(result)")
+    } catch {
+        print("Failed to sign in with error: \(error)")
+    }
+}
+```
+
+### Setup auth backend
```

Comment:
We should add a section on how this works when we have auth type in amplifyconfiguration.json

Reply:
I can do that as a follow up